### PR TITLE
Adding a check for missing resource

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Plus making sure the resource look up works for runtimes with no access to internal types

Plus removes dots from resource name as this is causing problems with msbuild on OSX. See Microsoft/msbuild#541

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
@joshfree

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs [hunk 1]
```diff
@@ -29,13 +29,19 @@ internal unsafe static class UnicodeHelpers
...
         [MethodImpl(MethodImplOptions.NoInlining)]
         private static uint[] CreateDefinedCharacterBitmap()
         {
             // The stream should be exactly 8KB in size.
-            var stream = typeof(UnicodeHelpers).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode-8.0.0-defined-characters.bin");
+            var stream = typeof(UnicodeRange).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode8definedcharacters.bin");
+
+            if (stream == null)
+            {
+                throw new BadImageFormatException();
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs]: We're fail-fasting below if the stream is of the wrong length or if we're unable to read all of the data.  Why the different treatment here?

Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs]: I just copied what System.Text.Encodings.CodePages does with its data resource stream, i.e. it throws an exception.

Moreover, I am not sure we actually want to fail fast here in the first place. Thoughts?

Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs]: I'm fine with the exception instead.  It just seems like we should be consistent with either throwing or fail-fasting if the image is malformed.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
```
CSC : error CS1566: Error reading resource 'System.Text.Encodings.Web.Resources.unicode-8.0.0-defined-characters.bin' -- 'Could not find file 'd:\j\workspace\windows_nt_de---06b7984d\src\System.Text.Encodings.Web\src\Resources\unicode-8.0.0-defined-characters.bin'.' [d:\j\workspace\windows_nt_de---06b7984d\src\System.Text.Encodings.Web\src\System.Text.Encodings.Web.csproj]
```

**Comment by KrzysztofCwalina** [comment]
dotnet-bot test outerloop OSX debug

**Comment by mellinoe** [comment]
@dotnet-bot test outerloop OSX debug

(The outer loop job ran on dci-macpro-05 which hasn't been patched yet :disappointed: )

**Comment by mellinoe** [comment]
@dotnet-bot Test OuterLoop OSX Debug.

Forgot to restart dci-macpro-11, so it had the wrong ulimit configuration...

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this please

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this please

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs]: I actually think there is a difference between these. The resource might be missing for configuration reasons. e.g. satellite assembly missing. The other checks can only fail if we produced the stream incorrectly or it got corrupted on disk.

Also, I was trying to minimize the churn in this change. I don't think this is a good time to change the rest of the routine.

**Comment by KrzysztofCwalina** [comment]
@mellinoe, could you review?

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs]: That's fine.  Up to you.

**Comment by stephentoub** [comment]
Test OuterLoop OSX Debug please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
Test OuterLoop OSX Debug please

**Pull request merged**